### ISSUE TITLE
Duplicate packages on deployed reaction server (tarball on digital ocean)

### ISSUE THREAD

**Issue by spencern** [reactioncommerce/reaction#740]
Today we updated our development server from Reaction 0.9.5 to Reaction dev (0.11.x I think?). Everything was running perfectly in our local environment, but we were running 0.11.x in a clean directory on the local and we were overwriting a 0.9.5 install on our development server.

We were able to get everything working on the development server, but from our first server restart after updating bundle we had multiple of each package showing up in our dashboard.

Here's some screenshots showing what I mean.

<img width="372" alt="screen shot 2016-01-28 at 3 05 51 pm" src="https://cloud.githubusercontent.com/assets/1203639/12660891/91c0a67c-c5d3-11e5-9df5-24ba13fa9f9d.png">

<img width="283" alt="screen shot 2016-01-28 at 3 06 31 pm" src="https://cloud.githubusercontent.com/assets/1203639/12660889/91be46e8-c5d3-11e5-8532-6d4c7191dd7b.png">

<img width="1464" alt="screen shot 2016-01-28 at 3 06 55 pm" src="https://cloud.githubusercontent.com/assets/1203639/12660890/91bf649c-c5d3-11e5-9e00-50a1ddee43d2.png">

Additionally after digging a little deeper, we discovered that restarting our server caused a third, (and an additional restart caused a fourth duplicate) of each package to show up.

**Of Note:**
* We had at least one package that was not previously installed on this server until we updated it to 0.11.x and that package was installed only once initially.

* We tried nuking our packages directory by dropping directly into a mongo shell and running `db.Packages.remove({})` - when we restarted the server after that, it loaded 2 of every package.

```
INFO core: JobServer started: true
INFO core: Modified 0 documents while importing 21 to Translations
INFO core: Modified 0 documents while importing 1 to Shops
INFO core: Modified 0 documents while importing 1 to Products
INFO core: Modified 0 documents while importing 2 to Tags
INFO core: Initializing GetOutfitted Dev Droplet core
INFO core: Initializing REACTION core
INFO core: Initializing GetOutfitted Dev Droplet reaction-accounts
INFO core: Initializing REACTION reaction-accounts
INFO core: Initializing GetOutfitted Dev Droplet reaction-inventory
INFO core: Initializing REACTION reaction-inventory
INFO core: Initializing GetOutfitted Dev Droplet reaction-shipping
INFO core: Initializing REACTION reaction-shipping
INFO core: Initializing GetOutfitted Dev Droplet reaction-analytics
INFO core: Initializing REACTION reaction-analytics
INFO core: Initializing GetOutfitted Dev Droplet reaction-social
INFO core: Initializing REACTION reaction-social
INFO core: Initializing GetOutfitted Dev Droplet reaction-paymentmethod
INFO core: Initializing REACTION reaction-paymentmethod
INFO core: Initializing GetOutfitted Dev Droplet reaction-paypal
INFO core: Initializing REACTION reaction-paypal
INFO core: Initializing GetOutfitted Dev Droplet reaction-stripe
INFO core: Initializing REACTION reaction-stripe
INFO core: Initializing GetOutfitted Dev Droplet rental-products
INFO core: Initializing REACTION rental-products
INFO core: Initializing GetOutfitted Dev Droplet reaction-advanced-fulfillment
INFO core: Initializing REACTION reaction-advanced-fulfillment
INFO core: Initializing GetOutfitted Dev Droplet reaction-shopify-products
INFO core: Initializing REACTION reaction-shopify-products
INFO core: Initializing GetOutfitted Dev Droplet reaction-shopify-orders
INFO core: Initializing REACTION reaction-shopify-orders
INFO core: Initializing GetOutfitted Dev Droplet reaction-local-delivery
INFO core: Initializing REACTION reaction-local-delivery
INFO core: Initializing GetOutfitted Dev Droplet reaction-zopim
INFO core: Initializing REACTION reaction-zopim
**INFO core: Modified 30 documents while importing 30 to Packages**
INFO core: Reaction initialization finished.
INFO core: Modified 1 document while importing 1 to Shipping
```

It mentions that it has `Modified 30 documents while importing 30 to Packages` but we have exactly 15 packages we are trying to load. I wonder if this is because we have two shops that are somehow both loading?

I'll update this issue as we debug this and let people know if we figure it out.

**Comment by aaronjudd** [comment]
This sounds like it's related to https://github.com/reactioncommerce/reaction/issues/693 maybe?

**Comment by spencern** [comment]
Definitely related.

Also worth mentioning, some gitter chat history from @tdecaluwe and I

> tdecaluwe 15:37
> spencern, I've been seeing this as well, do you have multiple shops in your db?
> 
> spencern 15:38
> Not intentionally, but it seems as if perhaps updating to 0.11.x added a second shop. goes to look
> that may be the issue, we have 2 shops now
> tdecaluwe do you know if theres a way to admin shops from the dashboard or should I just delete the additional shop via mongo shell?
> 
> tdecaluwe 15:42
> That's the same thing I've been seeing. This is related to the lack of proper multiple shop/tenant support. However, a provisional fix for this would be good to have. If you uninstall the reaction-sample-data package you should be fine right now.
> 
> spencern 15:43
> ah, that must have been the issue is dropping that package on top of our existing install

Seems like it's 100% caused by having multiple shops present.

**Comment by aaronjudd** [comment]
You should have one package entry for each shop-> 
```
INFO core: Initializing REACTION reaction-shopify-orders
INFO core: Initializing GetOutfitted Dev Droplet reaction-local-delivery
```
That's the correct behaviour, but the UI is the real issue. We never developed it to deal with multi shop, but all it shouldn't be that tough. 
I was just searching (without success) but there should be an issue tracking mult-shop admin @mikemurray and I have been discussing this, and agree it should be a priority.  

I'm actually sort of surprised you didn't see this on 0.9x, that's probably because we didn't update the records on every restart, only the first time setup and if that was done with only one shop record...  So if you only had one set of packages, that's the way it would have stayed.

**Comment by aaronjudd** [comment]
@tdecaluwe I think we should add a second shop to the reaction-sample-data for 0.12 -> that way we can fixing the multi-store (by force if needed)

**Comment by spencern** [comment]
I think the reason we didn't see this on 0.9x was because we never had more than one shop on our server at one point.

I agree that the real issue is the UI, I think that what was really the heart of the confusion for us is that an 'example' shop was added to an existing install that already had a shop.

We were able to undo everything that the `reaction-sample-data` package added with these steps:
we were able to undo all of this by:
1. following @tdecaluwe’s suggestion and commenting out reaction-sample-data
2. rebuilding our tarball and uploading it to our server
3. Removing the offending shop via mongo shell (db.Shops.remove({id_: “ReactionShopId”})
4. Removing all packages from our server (db.Packages.remove({}))
5. Restarting our server to re-initialize all packages
6. Adding all of our keys and other info back into our packages

This allowed us to complete an upgrade of an existing install without losing Products, Orders, Users, etc even after having some weirdness with the duplicate packages and extra shop.

I'd suggest that perhaps `reaction-sample-data` should check to see if there are any existing shops present and if so, not add any additional shop data. That way it could work by default for new setups, but also install seamlessly over the top of existing installs. (This might also fix the issue where restarting a server with reaction-sample-data installed causes a new shop to be created and duplicates all of the packages)

Perhaps it's also more of a doc issue and I should have just known to disable `reaction-sample-data` before installing. Either way, not a huge deal, it only cost us 30-45 minutes and didn't harm any of our shop data.
